src/core/exception.d

```diff
@@ -371,6 +371,70 @@ unittest
     }
 }
 
+/**
+ * Thrown on an array misalignment error, that may occur during an array reinterpreting case.
+ */
+class ArrayCastMisalignmentError : Error
+{
+    this(string fromType, size_t fromSize, string toType, size_t toSize,
+        string file = __FILE__, size_t line = __LINE__, Throwable next = null) @trusted pure nothrow @nogc
+    {
+        import core.internal.string : unsignedToTempString;
+        import core.stdc.stdlib : alloca;
+
+        const(char)[][8] msgComponents =
```

Comment:
How about this for formulating a more detailed error message.  I'm trying to avoid using the C standard library ([`alloca` actually isn't part of the C standard library](https://github.com/dlang/druntime/blob/9a8edfb48e4842180c706ee26ebd8edb10be53f4/src/rt/alloca.d#L58)).

Reply:
> I'm trying to avoid using the C standard library 

Why ?
Also, I would suggest to just store those values as field and do the formatting in `toString`

Reply:
> Why ?

Because eventually I'd like to jettison it completely from druntime.  I also need to consider using the same code for the `assert` message in `__ArrayCast`.

Reply:
I need to come up with a utility for creating a dynamic string on the stack.

Reply:
I ended up merging the message generation code into the `__ArrayCast` implementation itself so I could use it for the `assert` as well.